services/galley/src/Galley/API/Util.hs

```diff
@@ -466,22 +455,31 @@ fromRegisterConversation (Qualified usr localDomain) MkRegisterConversation {..}
         cnvReceiptMode = rcReceiptMode
       }
   where
-    me :: Map Domain [Public.Member] -> Galley Public.Member
-    me m = case Map.lookup localDomain m >>= find ((usr ==) . Member.memId) of
-      Nothing -> throwM convMemberNotFound
-      Just v -> pure v
-    others :: Map Domain [Public.Member] -> [OtherMember]
-    others =
-      Map.foldlWithKey' (\acc d mems -> fmap (memToOther d) mems <> acc) []
-        -- make sure not to include 'usr' in the list of others
-        . Map.adjust (filter ((usr /=) . Public.memId)) localDomain
-    memToOther :: Domain -> Member.Member -> OtherMember
-    memToOther d mem =
-      OtherMember
-        { omQualifiedId = Qualified (Member.memId mem) d,
-          omService = Member.memService mem,
-          omConvRoleName = Member.memConvRoleName mem
-        }
+    -- Currently this function creates a Member with default conversation attributes
+    -- FUTUREWORK(federation): retrieve member's conversation attributes (muted, archived, etc) here once supported by the database schema. 
+    me :: Set OtherMember -> Galley Public.Member
+    me s =
+      case find ((== qusr) . omQualifiedId) . Set.toList $ s of
+        Nothing -> throwM convMemberNotFound
+        Just v ->
+          pure
+            Public.Member
+              { memId = qUnqualified . omQualifiedId $ v,
+                memService = omService v,
+                memOtrMuted = False,
+                memOtrMutedStatus = Nothing,
+                memOtrMutedRef = Nothing,
+                memOtrArchived = False,
+                memOtrArchivedRef = Nothing,
+                memHidden = False,
+                memHiddenRef = Nothing,
+                memConvRoleName = omConvRoleName v
+              }
+    others :: Set OtherMember -> [OtherMember]
```

Comment:
Here it would also be nicer to use a higher-level fold (i.e. `foldMap` like above).

Reply:
Done!